### ISSUE TITLE
Better favicon inclusion (OSOE-652)

### ISSUE THREAD

**Issue by Piedone** [Lombiq/Orchard-Base-Theme#77]
Currently, `~/favicon.ico` is included at all times. This necessitates all child themes to add a redirect, [like the sample does](https://github.com/Lombiq/Orchard-Base-Theme/blob/58317019f449c5bed68dcb3b90e55b721996ae60/Lombiq.BaseTheme.Samples/Startup.cs#L39), which is not just a chore but the redirect also makes UX worse. Instead, do the following:

- [x] Similar to [what PGBBT does](https://github.com/Lombiq/Pretty-Good-Bootstrap-Base-Theme/blob/c693225f217078eb28af4334c21e3491075d1d01/Views/Resources.cshtml#L20), include the favicon from a conventional location of the child theme.
- [ ] Use a cached file existence check for the file, and only include it if it exists.
- [ ] Provide a solution for [Media Themes](https://github.com/Lombiq/Hosting-Media-Theme) that derive from Base Theme but aren't standard themes. How should you include a favicon there? Provide the necessary code and/or documentation for this. Ideally, the same convention than for standard themes would work.
- [x] Remove the now unnecessary redirect from the sample.

[Jira issue](https://lombiq.atlassian.net/browse/OSOE-652)

**Comment by sarahelsaig** [comment]
What I did so far was to add a site setting with a media field editor to pick out the icon:
![image](https://github.com/Lombiq/Orchard-Base-Theme/assets/4007293/c650ae64-3b5c-415a-89f9-c009134c3354)
During saving its media path is stored along with a timestamp, so if you replace the file and save the setting again that will correctly bust the cache. The icon link is only added if the site setting is not null.

This makes the costly file existence check unnecessary, because the only way that can fail is if you set the icon in the site setting and then remove the file from the media store. But that would be a user error that's fairly hard to do on accident so not our responsibility. 

I don't think we actually have to do anything special for media themes, although I don't have much experience with that so please tell me if there is something I missed. Since the icon is injected using a resource filter, it works if I switch over to the "Lombiq Hosting - Media Theme" as long as the Layout template has a `Header` section and the "Lombiq Base Theme" is enabled. But I see that `Lombiq.BaseTheme` isn't a dependency of `Lombiq.Hosting.MediaTheme` so I'm not sure if this is the right approach?

**Comment by Piedone** [comment]
Having an admin option to add the favicon is great, and nice to have especially on DotNest, but we need support for including the file directly when just working with Base Theme from code.

Media Themes can have their base theme set dynamically: https://github.com/Lombiq/Hosting-Media-Theme/blob/0346996e97f3a5208515b003f39a46603471b00e/Lombiq.Hosting.MediaTheme.Bridge/Services/ExtensionManagerDecorator.cs#L77

BTW the file existence check mentioned in the issue needn't be costly due to it being cached. The Media module's cache busting does the same thing, it also needs to access the file to generate its hash but only once.

**Comment by sarahelsaig** [comment]
> but we need support for including the file directly when just working with Base Theme from code.

What do you have in mind? Currently you can include it via a setup or migration recipe like [here](https://github.com/Lombiq/Orchard-Base-Theme/blob/28698a119ab0cfd5dab3dd5f6d380818d9b0de81/Lombiq.BaseTheme.Samples/Migrations/Lombiq.BaseTheme.Samples.UpdateFrom0.recipe.json), which I feel is straight forward enough. Otherwise the developer can just opt to make a custom resource filter [and register a link resource](https://github.com/Lombiq/Helpful-Libraries/blob/704e3380212256184035afee21e78907a73377e2/Lombiq.HelpfulLibraries.OrchardCore/ResourceManagement/ResourceFilter.cs#L47) now that the favicon link coming from the base theme is optional. But I wouldn't bother with that when admin setting is fairly easy to configure and more user friendly on the long term.

**Comment by Piedone** [comment]
What I mentioned in the issue: include the favicon from a conventional location of the child theme.
